tests/test_pipeline/test_pipeline.py

```diff
@@ -408,84 +361,231 @@ def test_get_metrics_interface(
-        (3, 3, 1),
-        (3, 4, 1),
-        (4, 1, 4),
-        (4, 2, 2),
-        (4, 3, 2),
-        (4, 4, 1),
-        (4, 5, 1),
+        ("expand", 3, True, None, [0, 0, 0], [-22, -15, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 3, True, 1, [0, 0, 0], [-10, -9, -8], [-9, -8, -7], [-3, -2, -1]),
+        ("expand", 3, True, 10, [0, 0, 0], [-28, -18, -8], [-27, -17, -7], [-21, -11, -1]),
+        ("expand", 3, False, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 3, False, 1, [0, 0, 0], [-10, -10, -10], [-9, -8, -7], [-3, -2, -1]),
+        ("expand", 3, False, 10, [0, 0, 0], [-28, -28, -28], [-27, -17, -7], [-21, -11, -1]),
+        ("expand", 1, 1, None, [0], [-8], [-7], [-1]),
+        ("expand", 1, 2, None, [0], [-8], [-7], [-1]),
+        ("expand", 3, 1, None, [0, 0, 0], [-22, -15, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 3, 2, None, [0, 0, 0], [-22, -22, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 3, 3, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 3, 4, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
+        ("expand", 4, 1, None, [0, 0, 0, 0], [-29, -22, -15, -8], [-28, -21, -14, -7], [-22, -15, -8, -1]),
+        ("expand", 4, 2, None, [0, 0, 0, 0], [-29, -29, -15, -15], [-28, -21, -14, -7], [-22, -15, -8, -1]),
+        ("expand", 4, 2, 1, [0, 0, 0, 0], [-11, -11, -9, -9], [-10, -9, -8, -7], [-4, -3, -2, -1]),
+        ("expand", 4, 2, 10, [0, 0, 0, 0], [-38, -38, -18, -18], [-37, -27, -17, -7], [-31, -21, -11, -1]),
+        ("expand", 4, 3, None, [0, 0, 0, 0], [-29, -29, -29, -8], [-28, -21, -14, -7], [-22, -15, -8, -1]),
+        ("expand", 4, 4, None, [0, 0, 0, 0], [-29, -29, -29, -29], [-28, -21, -14, -7], [-22, -15, -8, -1]),
+        ("expand", 4, 5, None, [0, 0, 0, 0], [-29, -29, -29, -29], [-28, -21, -14, -7], [-22, -15, -8, -1]),
+        ("constant", 3, True, None, [0, 7, 14], [-22, -15, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("constant", 3, True, 1, [0, 1, 2], [-10, -9, -8], [-9, -8, -7], [-3, -2, -1]),
+        ("constant", 3, True, 10, [0, 10, 20], [-28, -18, -8], [-27, -17, -7], [-21, -11, -1]),
+        ("constant", 3, False, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
+        ("constant", 3, False, 1, [0, 0, 0], [-10, -10, -10], [-9, -8, -7], [-3, -2, -1]),
+        ("constant", 3, False, 10, [0, 0, 0], [-28, -28, -28], [-27, -17, -7], [-21, -11, -1]),
+        ("constant", 1, 1, None, [0], [-8], [-7], [-1]),
+        ("constant", 1, 2, None, [0], [-8], [-7], [-1]),
+        ("constant", 3, 1, None, [0, 7, 14], [-22, -15, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("constant", 3, 2, None, [0, 0, 14], [-22, -22, -8], [-21, -14, -7], [-15, -8, -1]),
+        ("constant", 3, 3, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
+        ("constant", 3, 4, None, [0, 0, 0], [-22, -22, -22], [-21, -14, -7], [-15, -8, -1]),
```

Comment:
Maybe  these parameters  could be more obvious in case we use `horizon` as variable

Reply:
Ok, I'll extract it.